sampler.go

```diff
@@ -438,7 +477,23 @@ func (s *RemotelyControlledSampler) updateSampler() {
 }
 
 func (s *RemotelyControlledSampler) extractSampler(res *sampling.SamplingStrategyResponse) (Sampler, error) {
-	// TODO create adaptive sampler
+	if strategies := res.GetOperationSamplingStrategies(); strategies != nil {
+		samplingRates := make(map[string]float64)
+		for _, strategy := range strategies {
+			samplingRates[strategy.Operation] = strategy.ProbabilisticSampling.SamplingRate
+		}
+		if sampler, ok := s.sampler.(*adaptiveSampler); ok {
+			s.Lock()
+			sampler.update(defaultLowerBound, defaultSamplingProbability, samplingRates)
+			s.Unlock()
+			return sampler, nil
+		}
+		sampler, err := NewAdaptiveSampler(defaultLowerBound, defaultSamplingProbability, samplingRates)
```

Comment:
struct SamplingStrategyResponse {
    1: required SamplingStrategyType strategyType -> ignore
    2: optional ProbabilisticSamplingStrategy probabilisticSampling -> defaultSamplingProbability
    3: optional RateLimitingSamplingStrategy rateLimitingSampling -> defaultLowerBound
    4: optional list<OperationSamplingStrategy> operationSamplingStrategies -> per operation sampling probability
}

Lower value made no sense in my previous comment